tests/upgrade-tests/utils.py

```diff
@@ -41,10 +41,11 @@ def dump_model_info(model):
 def assert_no_unit_errors(model):
     for unit in model.units.values():
         workload_status = unit.data['workload-status']['current']
-        juju_status = unit.data['juju-status']['current']
         assert workload_status != 'error'
-        assert juju_status != 'failed'
-        assert juju_status != 'lost'
+        if 'juju-status' in unit.data:
```

Comment:
This seems to silence the error. Could this be shadowing any real problem? Do you fancy a retry before giving up with a message in the logs saying we gave up because we did not get juju status?